requirements/customer/cli.tex

```diff
@@ -0,0 +1,102 @@
+\newcommand{\jacommand}[3]{
+
+\textsc{\large{\textbf{#1}}}
+\begin{itemize}
+\item[\textbf{Options:}] #2
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\newcommand{\jaoption}[3]{
+\begin{itemize}
+\item[] \textbf{#1} (#2)
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\subsection{Command Line Interface}
+All components of JobAdder are accessible through a command line interface (CLI).
+The command line interface on a machine with JobAdder installed has the following structure:
+\begin{equation}
+\text{<component name> <command> <options> <targets>},
+\end{equation}
+where the \textit{component name} can be any one of the following:
+\begin{itemize}
+\item \textbf{jobadder} to access the \textit{user client}.
+\item \textbf{jobcenter} to access the \textit{server}.
+\item \textbf{jobworker} to access the \textit{worker client}.
+\end{itemize}
+Commands, options, and valid targets are and options are specified in the following sections.
+
+Options can be added in the following ways:
+\begin{itemize}
+\item By supplying their names like "--<option\_x> --<option\_y>=2 --<option\_z>".
+\item By supplying their abbreviations like "-x -y 2 -z".
+\item By supplying their abbreviations like "-xz -y 2".
+\end{itemize}
+
+The available \textit{commands} are generally specific to the used component.
+Likewise, the available \textit{options} and \textit{targets} are generally specific to the commands of a used component.
+However, some commands and options are shared between components or commands.
+
+The JobAdder CLI ignores capitalization.
+JobAdder CLI options override the values of configuration files.
+If no command is given, the general help text for the given component is printed out.
+\subsubsection{Shared Options}
+\jaoption{config}{c}{Overrides the global configuration file with the given configuration file.}
+\jaoption{detach}{d}{Detaches the execution of the given command from the command line.}
```

Comment:
I don't understand what this function is about, can you elaborate?
Do you mean that it makes the application run in the background? (If yes, `run in the background` is much more common wording for this AFAIK)

Reply:
I think that one of the simplest ways of debugging a job would be to forward the command line output of the job to the terminal that the job was added on.
I think that this should be the default behavior.
If the detach option is given the command line should immediately return rather than waiting for the job to terminate.
If we are indeed talking about the same thing I have no problem with renaming the option to something like background.

Reply:
Oh yeah, that's what I'm talking about too :)
Now that I think about it, I'm not very sure what the proper wording is, maybe somebody else can say what they think is more common - `detach from terminal` or `run in the background/as daemon`

Reply:
I'm for `run in the background/as daemon`. I think it suits better a requirements paper.